src/config.js

```diff
@@ -1,121 +1,89 @@
 'use strict'
 
 const Key = require('interface-datastore').Key
-const queue = require('async/queue')
-const waterfall = require('async/waterfall')
-const _get = require('dlv')
-const _set = require('just-safe-set')
-const Buffer = require('buffer').Buffer
+const Queue = require('p-queue')
+const _get = require('lodash.get')
+const _set = require('lodash.set')
+const _has = require('lodash.has')
+const Buffer = require('safe-buffer').Buffer
 
 const configKey = new Key('config')
 
 module.exports = (store) => {
-  const setQueue = queue(_doSet, 1)
+  const setQueue = new Queue({ concurrency: 1 })
 
   const configStore = {
     /**
      * Get the current configuration from the repo.
      *
      * @param {String} key - the config key to get
-     * @param {function(Error, Object)} callback
-     * @returns {void}
+     * @returns {Promise<Object>}
      */
-    get (key, callback) {
-      if (typeof key === 'function') {
-        callback = key
-        key = undefined
-      }
+    async get (key) {
       if (!key) {
         key = undefined
       }
-      store.get(configKey, (err, encodedValue) => {
-        if (err) { return callback(err) }
-
-        let config
-        try {
-          config = JSON.parse(encodedValue.toString())
-        } catch (err) {
-          return callback(err)
-        }
-
-        if (typeof key === 'undefined') {
-          return callback(null, config)
-        }
 
-        if (typeof key !== 'string') {
-          return callback(new Error('Key ' + key + ' must be a string.'))
-        }
-
-        const value = _get(config, key, null)
-
-        if (value === null) {
-          return callback(new Error('Key ' + key + ' does not exist in config.'))
-        }
+      const encodedValue = await store.get(configKey)
+      const config = JSON.parse(encodedValue.toString())
+      if (key !== undefined && !_has(config, key)) {
+        throw new Error(`Key ${key} does not exist in config`)
```

Comment:
Should we add some tests for these error cases?

Reply:
I have increased the test coverage.  The only branches that are not covered now stem from [this line](https://github.com/ipfs/js-ipfs-repo/blob/d52ecc09bd0c69dbab439c559624f3d2d223edac/src/blockstore.js#L175) which I'm not sure how to test, given that the `cid` passed to it has already been run through `CID.isCID()` so nothing in that function should throw.

Reply:
I have increased the test coverage.